File: Data Structures & Algorithms/maximum-subarray/submission-0.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curSum = 0
        curMax = nums[0]

        for num in nums:
            if curSum < 0:
                curSum = 0
            curSum += num
            curMax = max(curMax, curSum)
        
        return curMax

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> seen = new HashSet<>();

        for (int num : nums) {
            seen.add(num);
        }

        int curMax = 0;

        for (int num : nums) {
            if (!seen.contains(num-1)) {
                int length = 1;
                while (seen.contains(num + length)) {
                    length++;
                }
                curMax = Math.max(curMax, length);
            }
        }

        return curMax;

    }
}
